paddle/fluid/framework/ir/subblock_to_graph_pass.cc

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/subblock_to_graph_pass.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+void TransformSubblockToProgram(framework::BlockDesc* block_desc,
+                                proto::ProgramDesc* program_desc) {
+  *program_desc->add_blocks() = *block_desc->Proto();
+}
+
+std::unique_ptr<ir::Graph> SubblockToGraphPass::ApplyImpl(
+    std::unique_ptr<ir::Graph> graph) const {
+  // Init sub graph
+  if (!graph->Has(kSubblockGraphAttr)) {
+    graph->Set(kSubblockGraphAttr, new subgraphs_t);
+  }
```

Comment:
else return or throw error?

Reply:
Just init here, create a subgraphs if not exists. No need to raise error.

Reply:
Are you creating the subgraphs if exists? (graph->Has returns true?)

Reply:
done

https://github.com/PaddlePaddle/Paddle/pull/14936/files#diff-f4c0665c5674c253c8aa0179cf6022f4R33